fs/remote/resolver.go

```diff
@@ -554,6 +387,84 @@ func (f *httpFetcher) isSingleRangeMode() bool {
 	return r
 }
 
+// redirect sends a GET request to a given endpoint with a given http.RoundTripper and
+// returns the final URL in a redirect chain.
+func redirect(ctx context.Context, blobURL string, tr http.RoundTripper) (string, error) {
```

Comment:
If we get another redirection after following the chain (10 redirects) than we return the redirect URL. Meaning, although we haven't found the root we have at-least gotten alot closer. 

https://github.com/awslabs/soci-snapshotter/pull/977/files#diff-bbd28009e6b9883e1795e398bdac4a1fef4ad04d241291152f43f9f6546ac120R415-R417

Reply:
That makes sense, but would it then be up to the calling function to decide what to do with it?

I initially asked this question because I was wondering if there's any chance we could try and finish resolving it ourselves, but 10 is already a ridiculous amount of redirects, so it's probably rarely fruitful to try even more

Reply:
> That makes sense, but would it then be up to the calling function to decide what to do with it?

It would take that to be the "real"/ fully redirected URL.

I suppose we could try, by sending another request with the redirect URL, but I don't expect that to be a common scenario, so I'm not sure it's worth it.

Reply:
Ah, right. I was not aware of that hence the:

```go
// If we still get a redirection, return the redirect URL. 
 if redir := res.Header.Get("Location"); redir != "" && res.StatusCode/100 == 3 { 
 	return redir, nil 
 } 
```
This path will never be hit, since we'll throw an error on too many redirects. I don't think this is a big deal but maybe we should add some sort of check to ensure the error wasn't TooManyRedirects. I'd rather return a URL than an error, WDYT?

Reply:
I think we should error as early as possible. We're trying to do a ranged get here on the URL for the layer so that we can see the final "real" URL so that we can skip the redirects in the future. If we get too many redirects here, we will also get too many redirects when we try to fetch spans. If we return a URL here, we're just pushing the failure point down the line.